pkg/notationutil/notationutil.go

```diff
@@ -0,0 +1,134 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package notationutil
```

Comment:
Can we combine this with `cosignutil` into a single package (e.g. `signutil`) and have a `notationutil.go` and a `cosignutil.go` within it.

I don't feel they will have more than one `go` files respectively. So one pkg per tool might not necessary.

Reply:
If I'm right [here](https://github.com/containerd/nerdctl/pull/2035#discussion_r1122636407), we will have `signNotation()` and `signCosign()` in one single package `signutil`. I'm good with that, but want to discuss the principle about "grouping pkg by functionalities" (one single signutil because they are both for signing) vs "grouping pkg by dependencies" (two utils because they are different dependencies).

I tried to find similar cases and found in Lima, there are separated pkgs for qemu and vz (virtualization framework wrapper), instead of one hypervisor pkg. Qemu and vz are both hypervisor dependencies of Lima while Cosign and Notation are both signing dependencies of nerdctl.

Reply:
Following up on this, I notice that both `cosignutil` and `notationutil` have the same dependencies (i.e., it's not like that `cosignutil` imports `github.com/sigstore/cosign` while `notationutil` imports `github.com/notary/notation`, then separating them _may_ reduce a dependency (e.g., `github.com/sigstore/cosign`) for a user if the user only imports `nerdctl/pkg/notationutil`)).

Given this, I don't see a practical benefit of separating them, and even in nerdctl the two packages are almost always imported together.

I'd suggest to combine them into a single pkg (e.g., `signutil`) unless other reviewers disagree. This also makes it easy to implement the refactor mentioned in https://github.com/containerd/nerdctl/pull/2035#discussion_r1123363560. (Otherwise, in which `pkg` to put the dispatcher function?)

Reply:
Sounds good.